src/Swarm/TUI/Controller.hs

```diff
@@ -401,28 +402,37 @@ stripCmd pty = pty
 
 -- | Handle a user input event for the REPL.
 handleREPLEvent :: AppState -> BrickEvent Name AppEvent -> EventM Name (Next AppState)
-handleREPLEvent s (VtyEvent (V.EvKey (V.KChar 'c') [V.MCtrl])) =
-  continue $
-    s
-      & gameState . robotMap . ix "base" . machine %~ cancel
-handleREPLEvent s (VtyEvent (V.EvKey V.KEnter [])) =
-  if not $ s ^. gameState . replWorking
-    then case processTerm' topTypeCtx topCapCtx entry of
-      Right t@(ProcessedTerm _ (Module ty _) _ _) ->
-        continue $
-          s
-            & uiState . uiReplForm %~ updateFormState ""
-            & uiState . uiReplType .~ Nothing
-            & uiState . uiReplHistory %~ addREPLItem (REPLEntry entry)
-            & uiState . uiError .~ Nothing
-            & gameState . replStatus .~ REPLWorking ty Nothing
-            & gameState . robotMap . ix "base" . machine .~ initMachine t topValCtx topStore
-            & gameState %~ execState (activateRobot "base")
-      Left err ->
-        continue $
-          s
-            & uiState . uiError ?~ txt err
-    else continueWithoutRedraw s
+handleREPLEvent s = \case
+  (VtyEvent (V.EvKey (V.KChar 'c') [V.MCtrl])) ->
+    let cancel' = do
+          cm <- let m = s ^. gameState . robotMap . to (! "base") . machine in cancel m
```

Comment:
Using `(! "base")` will crash if there is no robot named `"base"` which is a big no-no!  I will have to think about how to rewrite this code.

Reply:
Well I have no idea, let me know if you think of something :wink:

Reply:
I mean one simple thing is just to wrap this in a check: first check if the `robotMap` has a key `"base"` and if so do the cancelling stuff.